### ISSUE TITLE
Extending EscapeOutputSniff sniff makes tests fail

### ISSUE THREAD

**Issue by dingo-d** [WordPress/WordPress-Coding-Standards#1988]
## Bug Description
In my custom sniff, I'm extending the `WordPress.Security.EscapeOutput.OutputNotEscaped` sniff. I've added some modifications to it that shouldn't affect the native sniff behavior, but for some reason, when I'm running tests locally I get the following result

```bash
1) EightshiftCS\Eightshift\Tests\Security\CustomEscapeOutputUnitTest::testSniff
[LINE 190] Expected 1 error(s) in CustomEscapeOutputUnitTest.inc but found 0 error(s).
[LINE 191] Expected 2 error(s) in CustomEscapeOutputUnitTest.inc but found 0 error(s).
[LINE 196] Expected 0 error(s) in CustomEscapeOutputUnitTest.inc but found 1 error(s). The error(s) found were:
 -> All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'esc_form_field'. (Eightshift.Security.CustomEscapeOutput.OutputNotEscaped)
[LINE 197] Expected 0 error(s) in CustomEscapeOutputUnitTest.inc but found 1 error(s). The error(s) found were:
 -> All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'post_info'. (Eightshift.Security.CustomEscapeOutput.OutputNotEscaped)
[LINE 198] Expected 0 error(s) in CustomEscapeOutputUnitTest.inc but found 1 error(s). The error(s) found were:
 -> All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'cpt_info'. (Eightshift.Security.CustomEscapeOutput.OutputNotEscaped)
```

The test case is the same as in the original sniff + some additions that are passing the test.

I've tried to debug the sniff, and the `printingFunctions` variable is empty (in the test there is a set directive for `customPrintingFunctions` array).

The problem is worse when I've run the sniff through my CI/CD

https://github.com/infinum/eightshift-coding-standards/pull/44/checks?check_run_id=2460374998

The addition to the sniff is just a small snippet inside the loop of the echo'd components

```php
/**
 * Eightshift libs addition
 */
$componentsFound = false;
if ($content === 'Components') {
	$componentsFound = true;
	$colon = $this->phpcsFile->findNext(Tokens::$emptyTokens, ($ptr + 1), null, true);

	if ($this->tokens[$colon]['code'] === \T_DOUBLE_COLON) {
		// This is a static call to a Components class method.
		$componentsMethodPtr = $this->phpcsFile->findNext(\T_STRING, ($colon + 1));

		// Check if the call is to a render method.
		if ($this->tokens[$componentsMethodPtr]['content'] === 'render') {
			$watch = true;
			continue;
		}
	}
}
```

Just checking if the `Components::render` exists and ignore it if it does. I've tried with and without the `$watch` parameter (not 100% sure what it does).

Also, when I run the sniff over the PHP example, the parts that are safe are flagged as unsafe

```bash
Run vendor/bin/phpcs --runtime-set ignore_warnings_on_exit 1 --report-full --report-checkstyle=./phpcs-report.xml
.....E 6 / 6 (100%)



FILE: Tests/RulesetCheck/RulesetTest8.inc
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 134 | ERROR | All output should be run through an escaping function
     |       | (see the Security sections in the WordPress Developer
     |       | Handbooks), found ''<div class="%s">''.
     |       | (Eightshift.Security.CustomEscapeOutput.OutputNotEscaped)
 140 | ERROR | All output should be run through an escaping function
     |       | (see the Security sections in the WordPress Developer
     |       | Handbooks), found ''</div>''.
     |       | (Eightshift.Security.CustomEscapeOutput.OutputNotEscaped)
----------------------------------------------------------------------
```

I've tried to debug the test, and I noticed that not only are `printingFunctions` not set, the `minimum_supported_version` was set to `5.1` even though it should be `5.4` when I'm running it over my custom code (it's set in the ruleset).

The only modifications are removed yoda conditions and I've reformatted it to be PSR12 compliant (naming and spacing mostly).

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.4.16
| PHP_CodeSniffer version | 3.6.0
| WPCS version            | 2.3.0
| WPCS install type       | Composer (custom ruleset)
| IDE (if relevant)       | Terminal on MacOs and PhpStorm 2021.1

## Tested Against `develop` branch?
- [ ] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by jrfnl** [comment]
@dingo-d I'm really sorry, but I have to be honest - IMO debugging a sniff in an external (commercial) standard is not really something which belongs in the WPCS repo, no matter that the sniff extends a WPCS sniff.
This can't be debugged without digging into the external standard and is more than likely a problem with that standard and not with the sniff as such as it works fine in the context of WPCS.

**Comment by dingo-d** [comment]
Yeah, I understand, I'll have to dig deeper into it to understand what is going on. It was just an odd behavior that happened when I just extended the sniff and removed the original from my ruleset, and some parts of it didn't work so I thought it was a bug in the WPCS.
